app/controllers/case_logs_controller.rb

```diff
@@ -20,12 +20,32 @@ def edit
     render :edit
   end
 
+  def update_checkbox_responses(case_log, questions_for_page)
+    result = {}
+    case_log.each do |question, answer|
+      if question == "previous_page"
+        result[question] = answer
+      elsif questions_for_page[question]["type"] == "checkbox"
+        questions_for_page[question]["answer_options"].keys.reject { |x| x.match(/divider/) }.each do |option|
+          result[option] = case_log[question].include?(option) ? true : false
+        end
+      else
+        result[question] = answer
+      end
```

Comment:
I'm wondering if there's a way we can shorten this? I'm noticing that the condition on like 26 gives the same result as the else on line 32